src/Sylius/Bundle/ApiBundle/Command/Checkout/ChoosePaymentMethod.php

```diff
@@ -15,64 +15,30 @@
 
 use Sylius\Bundle\ApiBundle\Command\IriToIdentifierConversionAwareInterface;
 use Sylius\Bundle\ApiBundle\Command\OrderTokenValueAwareInterface;
+use Sylius\Bundle\ApiBundle\Command\PaymentIdAwareInterface;
 use Sylius\Bundle\ApiBundle\Command\PaymentMethodCodeAwareInterface;
 
-class ChoosePaymentMethod implements OrderTokenValueAwareInterface, PaymentMethodCodeAwareInterface, IriToIdentifierConversionAwareInterface
+class ChoosePaymentMethod implements OrderTokenValueAwareInterface, PaymentIdAwareInterface, PaymentMethodCodeAwareInterface, IriToIdentifierConversionAwareInterface
 {
-    /** @var string|null */
-    public $orderTokenValue;
-
-    /**
-     * @immutable
-     *
-     * @var string|null
-     */
-    public $paymentId;
-
-    /**
-     * @immutable
-     *
-     * @var string|null
-     */
-    public $paymentMethodCode;
-
-    public function __construct(string $paymentMethodCode)
-    {
-        $this->paymentMethodCode = $paymentMethodCode;
+    public function __construct(
+        public string $paymentMethodCode,
+        public ?int $paymentId = null,
+        public ?string $orderTokenValue = null,
```

Comment:
```suggestion
        protected string $paymentMethodCode,
        protected mixed $paymentId,
        protected string $orderTokenValue,
```
as there is a possibility to create a command with all unnecessary data using context builder, we should use this and not allow the creation of an object with incomplete data